packages/python/plotly/_plotly_utils/colors/__init__.py

```diff
@@ -806,3 +806,54 @@ def named_colorscales():
     from _plotly_utils.basevalidators import ColorscaleValidator
 
     return [c for c in ColorscaleValidator("", "").named_colorscales]
+
+
+def sample_colorscale(colorscale, samplepoints, low=0.0, high=1.0, colortype="rgb"):
+    """
+    Samples a colorscale at specific points.
+
+    Interpolates between colors in a colorscale to find the specific colors
+    corresponding to the specified sample values. The colorscale can be specified
+    as a list of `[scale, color]` pairs, as a list of colors, or as a named
+    plotly colorscale. The samplepoints can be specefies an iterable of specific
+    points in the range [0.0, 1.0], or as an integer number of points which will
+    be spaced equally between the low value (default 0.0) and the high value
+    (default 1.0). The output is a list of colors, formatted according to the
+    specified colortype.
+    """
+    from bisect import bisect_left
+
+    try:
+        validate_colorscale(colorscale)
+    except exceptions.PlotlyError:
+        if isinstance(colorscale, str):
+            if colorscale in PLOTLY_SCALES:
+                colorscale = PLOTLY_SCALES[colorscale]
+            else:
+                raise exceptions.PlotlyError(
```

Comment:
I'm not sure I understand this error message... Under what circumstances would we accept "an rgb color or a hex color" here?